data-processing-lib/python/src/data_processing/runtime/pure_python/transform_orchestrator.py

```diff
@@ -74,15 +72,21 @@ def orchestrate(
         return 1
     # create additional execution parameters
     runtime = runtime_config.create_transform_runtime()
+    is_folder = issubclass(runtime_config.get_transform_class(), AbstractFolderTransform)
     try:
-        # Get files to process
-        files, profile, retries = data_access.get_files_to_process()
-        if len(files) == 0:
-            logger.error("No input files to process - exiting")
-            return 0
-        if retries > 0:
-            statistics.add_stats({"data access retries": retries})
-        logger.info(f"Number of files is {len(files)}, source profile {profile}")
+        if is_folder:
+            # folder transform
+            files = AbstractFolderTransform.get_folders(d_access=data_access)
```

Comment:
great minds, just moved it there

Reply:
As for data access, discussed it with Constantin, and it is too application specific

Reply:
This is a great idea. `get_folders()` cannot be in the `DataAccess`, because sometimes we know exactly the set of folders that we want to process (also the number of levels of subfolders that we want to take into consideration), and we don't need `DataAccess` to produce a list of those folders. But putting `get_folders()` in the runtime config will work and will provide maximum flexibility, because we can either use `DataAccess` to retrieve specific files, or we can just provide a list of folders ourselves.